mc-sema/bin_descend/get_cfg.py

```diff
@@ -449,6 +449,54 @@ def isExternalData(fn):
         return False
 
 
+def sanityCheckJumpTableSize(table_ea, ecount):
+    """ IDA doesn't correctly calculate some  jump table sizes """
+    if not isLinkedElf():
+        return ecount
+
+    if getBitness() != 64:
+        return ecount
+
+    table_insn = idautils.DecodeInstruction(table_ea)
+    if table_insn is None:
+        DEBUG("Could not decode instruction at {:x}\n".format(table_insn))
+        return ecount
+
+    if table_insn.Operands[0].type != idc.o_reg:
+        return ecount
+
+    DEBUG("Sanity checking table at {:x}\n".format(table_ea))
+
+    # get register we jump with
+    jmp_reg = table_insn.Operands[0].value
+
+    inst_ea = table_ea
```

Comment:
Agree, that should go in here. This is special casing a very specific type of jump table that we see in amd64 programs, that IDA fails to get the proper count of, in rare cases:

![image](https://cloud.githubusercontent.com/assets/1051838/18979705/6603f65e-869b-11e6-995b-bcbbaa3f3241.png)